dnn/variational_inference.ipynb

```diff
@@ -3,31 +3,47 @@
     {
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #57.        untransformed_var = krig_var + keras.layers.Lambda(lambda t: tf.math.log(1 + tf.exp(t)))(unscaled_var)
</pre><p>Using this technique to predict residuals is tricky... As written, this will force our predicted residuals to only be positive, which means we will only be able to predict <em>higher</em> variance than Kriging produces. I would suggest trying the following:</p><ul><li>Use this instead, which should have the intended effect (Pseudocode, please verify e.g. that you can pass two arguments pairwise to a lambda; otherwise, use the appropriate vector zipping transformation):<code>untransformed_var = keras.layers.Lambda(lambda t,k: tf.math.log(1 + tf.exp(t+k)))(unscaled_var, krig_var)﻿</code></li><li>Manually inspect the data going into the neural network (e.g. "inputs") to make sure it is what you expect, at least once</li><li>[Optional] Compare performance with vs. without variance boosting</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/165/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>